diff --git a/dist/src/index.d.ts b/dist/src/index.d.ts
index 51dfbbc2b6c8938ff4c7a4081a8c5467d6ac5af8..5ff7850e8466d01e220cfbb363c1284d0be79ef3 100644
--- a/dist/src/index.d.ts
+++ b/dist/src/index.d.ts
@@ -15,6 +15,7 @@ export * from "./render.js";
 export * from "./resource.js";
 export * from "./runtime/component.js";
 export * from "./runtime/intrinsic.js";
+export * from "./scheduler.js";
 export * from "./stc.js";
 export * from "./sti.js";
 export * from "./symbols/index.js";
diff --git a/dist/src/index.js b/dist/src/index.js
index e7a940344a718461d4c31eb7ace216fd735ddbcf..0870e7d14f09474b6765fade42374ee1b9ba1ae8 100644
--- a/dist/src/index.js
+++ b/dist/src/index.js
@@ -15,6 +15,7 @@ export * from "./render.js";
 export * from "./resource.js";
 export * from "./runtime/component.js";
 export * from "./runtime/intrinsic.js";
+export * from "./scheduler.js";
 export * from "./stc.js";
 export * from "./sti.js";
 export * from "./symbols/index.js";
@@ -22,5 +23,5 @@ export * from "./symbols/symbol-flow.js";
 export * from "./tap.js";
 export * from "./utils.js";
 export * from "./write-output.js";
-import "./debug.js";
+export * from "./debug.js";
 //# sourceMappingURL=index.js.map
\ No newline at end of file
diff --git a/dist/src/reactivity.js b/dist/src/reactivity.js
index 5c828765be3abec09ef2c1086f691385693419c4..de79c9858c8c8770d0d4aa743b7ba8265c1c74f3 100644
--- a/dist/src/reactivity.js
+++ b/dist/src/reactivity.js
@@ -3,10 +3,10 @@ import { scheduler } from "./scheduler.js";
 import { trace, TracePhase } from "./tracer.js";
 
 // check for multiple versions of alloy here.
-if (globalThis.__ALLOY__) {
-  throw new Error("Multiple versions of Alloy are loaded for this project. This will likely cause undesirable behavior.");
-}
-globalThis.__ALLOY__ = true;
+// if (globalThis.__ALLOY__) {
+//   throw new Error("Multiple versions of Alloy are loaded for this project. This will likely cause undesirable behavior.");
+// }
+// globalThis.__ALLOY__ = true;
 export function getElementCache() {
   return getContext().elementCache;
 }
diff --git a/src/index.ts b/src/index.ts
index 939ba3349b24361948b2275ae25c4a3704fc8d04..63b05e0fa5819e5fb53544e7b85d6a0f937f75eb 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -20,7 +20,7 @@ export {
   type ShallowReactive,
   type ShallowRef,
   type ToRef,
-  type ToRefs,
+  type ToRefs
 } from "@vue/reactivity";
 export * from "./binder.js";
 export * from "./code.js";
@@ -38,6 +38,7 @@ export * from "./render.js";
 export * from "./resource.js";
 export * from "./runtime/component.js";
 export * from "./runtime/intrinsic.js";
+export * from "./scheduler.js";
 export * from "./stc.js";
 export * from "./sti.js";
 export * from "./symbols/index.js";
diff --git a/src/reactivity.ts b/src/reactivity.ts
index 9e30999b39919206cd76d0181f586c18caf05b1d..3216a8b0b28cc4dfecb5789eb5e2a1b594228720 100644
--- a/src/reactivity.ts
+++ b/src/reactivity.ts
@@ -8,7 +8,7 @@ import {
   ShallowReactive,
   shallowRef,
   stop,
-  effect as vueEffect,
+  effect as vueEffect
 } from "@vue/reactivity";
 import type { RenderedTextTree } from "./render.js";
 import type { Children, ComponentCreator } from "./runtime/component.js";
@@ -17,12 +17,12 @@ import type { OutputSymbol } from "./symbols/output-symbol.js";
 import { trace, TracePhase } from "./tracer.js";
 
 // check for multiple versions of alloy here.
-if ((globalThis as any).__ALLOY__) {
-  throw new Error(
-    "Multiple versions of Alloy are loaded for this project. This will likely cause undesirable behavior.",
-  );
-}
-(globalThis as any).__ALLOY__ = true;
+// if ((globalThis as any).__ALLOY__) {
+//   throw new Error(
+//     "Multiple versions of Alloy are loaded for this project. This will likely cause undesirable behavior.",
+//   );
+// }
+// (globalThis as any).__ALLOY__ = true;
 
 export function getElementCache() {
   return getContext()!.elementCache;
@@ -107,7 +107,7 @@ export function root<T>(fn: (d: Disposable) => T, options?: RootOptions): T {
     takenSymbols: undefined,
     childrenWithContent: 0,
     isEmpty: ref(true),
-    isRoot: true,
+    isRoot: true
   };
 
   globalContext = context;
@@ -118,7 +118,7 @@ export function root<T>(fn: (d: Disposable) => T, options?: RootOptions): T {
         for (const d of context!.disposables) {
           d();
         }
-      }),
+      })
     );
   } finally {
     globalContext = globalContext!.owner;
@@ -136,7 +136,7 @@ export function untrack<T>(fn: () => T): T {
 
 export function memo<T>(fn: () => T, equal?: boolean): () => T {
   const o = shallowRef();
-  effect((prev) => {
+  effect(prev => {
     const res = fn();
     // eslint-disable-next-line @typescript-eslint/no-unused-expressions
     (!equal || prev !== res) && (o.value = res);
@@ -154,7 +154,7 @@ export function effect<T>(fn: (prev?: T) => T, current?: T) {
     takesSymbols: false,
     takenSymbols: undefined,
     childrenWithContent: 0,
-    isRoot: false,
+    isRoot: false
   };
 
   const cleanupFn = (final: boolean) => {
@@ -193,8 +193,8 @@ export function effect<T>(fn: (prev?: T) => T, current?: T) {
         trace(TracePhase.effect.trigger, () => {
           return `triggering ${isRef(event.target) ? `Ref:${refId(event.target)}` : event.target}, ${String(event.key)}`;
         });
-      },
-    },
+      }
+    }
   );
 }
 
@@ -235,13 +235,13 @@ export type CustomContextChildrenCallback = (child: Children) => void;
 const CUSTOM_CONTEXT_SYM = Symbol();
 
 export function createCustomContext(
-  useCallback: (useChildren: CustomContextChildrenCallback) => void,
+  useCallback: (useChildren: CustomContextChildrenCallback) => void
 ): CustomContext {
   return {
     [CUSTOM_CONTEXT_SYM]: true,
     useCustomContext(useCb: (children: Children) => void): void {
       useCallback(useCb);
-    },
+    }
   };
 }
 
