diff --git a/dist/src/index.d.ts b/dist/src/index.d.ts
index 51dfbbc2b6c8938ff4c7a4081a8c5467d6ac5af8..5ff7850e8466d01e220cfbb363c1284d0be79ef3 100644
--- a/dist/src/index.d.ts
+++ b/dist/src/index.d.ts
@@ -15,6 +15,7 @@ export * from "./render.js";
 export * from "./resource.js";
 export * from "./runtime/component.js";
 export * from "./runtime/intrinsic.js";
+export * from "./scheduler.js";
 export * from "./stc.js";
 export * from "./sti.js";
 export * from "./symbols/index.js";
diff --git a/dist/src/index.js b/dist/src/index.js
index e7a940344a718461d4c31eb7ace216fd735ddbcf..1e77e9f3f6490cfd10cda04c31a8d15904d11a81 100644
--- a/dist/src/index.js
+++ b/dist/src/index.js
@@ -15,6 +15,7 @@ export * from "./render.js";
 export * from "./resource.js";
 export * from "./runtime/component.js";
 export * from "./runtime/intrinsic.js";
+export * from "./scheduler.js";
 export * from "./stc.js";
 export * from "./sti.js";
 export * from "./symbols/index.js";
@@ -22,5 +23,5 @@ export * from "./symbols/symbol-flow.js";
 export * from "./tap.js";
 export * from "./utils.js";
 export * from "./write-output.js";
-import "./debug.js";
+export * from "./debug.js";
 //# sourceMappingURL=index.js.map
diff --git a/src/index.ts b/src/index.ts
index 939ba3349b24361948b2275ae25c4a3704fc8d04..63b05e0fa5819e5fb53544e7b85d6a0f937f75eb 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -20,7 +20,7 @@ export {
   type ShallowReactive,
   type ShallowRef,
   type ToRef,
-  type ToRefs,
+  type ToRefs
 } from "@vue/reactivity";
 export * from "./binder.js";
 export * from "./code.js";
@@ -38,6 +38,7 @@ export * from "./render.js";
 export * from "./resource.js";
 export * from "./runtime/component.js";
 export * from "./runtime/intrinsic.js";
+export * from "./scheduler.js";
 export * from "./stc.js";
 export * from "./sti.js";
 export * from "./symbols/index.js";
