diff --git a/dist/index.d.mts b/dist/index.d.mts
index c727c2c391ce42ed4958ade5a2617ff650f11d23..f3b5473331885d891e98b53c246e1cb37819f0a3 100644
--- a/dist/index.d.mts
+++ b/dist/index.d.mts
@@ -143,6 +143,12 @@ interface Config {
   }) => void;
   /** Custom generators */
   generators?: Record<string, Generator>;
+  /**
+   * Prefixes to identify automd blocks in markdown files.
+   *
+   * @default `["automd", "powerlines"]`
+   */
+  prefix?: string | string[];
 }
 declare const RESOLVED_CONFIG_SYMBOL: unique symbol;
 type ResolvedConfig = { [P in keyof Config]-?: Config[P] } & {
diff --git a/dist/index.mjs b/dist/index.mjs
index 62e86e4c1448e1fc390c22d1b7845d6f9d42bffc..8de13df8d3e34f9fab160241772de3d16254111a 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -508,9 +508,13 @@ var generators_default = {
 	contributors,
 	"dir-tree": dirTree
 };
-function findBlocks(md) {
+function findBlocks(md, config) {
 	const blocks = [];
-	for (const match of md.matchAll(/^(?<open><!--\s*automd:(?<generator>.+?)\s+(?<args>.*?)\s*-->)(?<contents>.+?)(?<close>^<!--\s*\/automd\s*-->)/gimsu)) {
+	for (const match of md.matchAll(new RegExp(`^(?<open><!--\s*(${
+            config.prefix.join("|")
+        }):(?<generator>.+?)\s+(?<args>.*?)\s*-->)(?<contents>.+?)(?<close>^<!--\s*\/(${
+            config.prefix.join("|")
+        })\s*-->)`, "gimsu"))) {
 		if (match.index === void 0 || !match.groups) continue;
 		const start = match.index + match.groups.open.length || 0;
 		const end = start + match.groups.contents.length;
@@ -530,8 +534,8 @@ function findBlocks(md) {
 	}
 	return blocks;
 }
-function containsAutomd(md) {
-	return /^<!--\s*automd:/gimsu.test(md);
+function containsAutomd(md, config) {
+	return new RegExp(`^<!--\\s*(${config.prefix.join("|")}):`, "gimsu").test(md);
 }
 function parseRawArgs(rawArgs) {
 	const args = Object.create(null);
@@ -554,6 +558,10 @@ function resolveConfig(config) {
 		[RESOLVED_CONFIG_SYMBOL]: true,
 		...config
 	};
+    _config.prefix = _config.prefix || [];
+    if (typeof _config.prefix === "string") _config.prefix = [_config.prefix];
+    if (!_config.prefix.includes("automd")) _config.prefix.push("automd");
+    if (!_config.prefix.includes("powerlines")) _config.prefix.push("powerlines");
 	_config.dir = resolve(_config.dir);
 	_config.input = (Array.isArray(_config.input) ? _config.input : [_config.input]).filter(Boolean);
 	return _config;
@@ -586,7 +594,7 @@ async function transform(contents, _config, url) {
 		...generators_default,
 		...config.generators
 	};
-	const blocks = findBlocks(contents);
+	const blocks = findBlocks(contents, config);
 	for (const block of blocks) {
 		const result = await _transformBlock(block, config, generators, url);
 		if (result.unwrap) editor.overwrite(block._loc.start, block._loc.end, `${result.contents.trim()}`);
@@ -642,6 +650,11 @@ async function _transformBlock(block, config, generators, url) {
 	}
 }
 async function automd(_config = {}) {
+    _config.prefix = _config.prefix || [];
+    if (typeof _config.prefix === "string") _config.prefix = [_config.prefix];
+    if (!_config.prefix.includes("automd")) _config.prefix.push("automd");
+    if (!_config.prefix.includes("powerlines")) _config.prefix.push("powerlines");
+
 	const start = performance.now();
 	const config = await loadConfig(_config.dir, _config);
 	let inputFiles = config.input;
